RevenueCatUI/Scripts/Platforms/Android/AndroidCustomerCenterPresenter.cs

```diff
@@ -0,0 +1,295 @@
+        {
+            _storedCallbacks?.OnRestoreStarted?.Invoke();
+        }
+
+        private void OnRefundRequestStarted(string productIdentifier)
+        {
+            _storedCallbacks?.OnRefundRequestStarted?.Invoke(
+                new RefundRequestStartedEventArgs(productIdentifier)
+            );
+        }
+
+        private void OnRefundRequestCompleted(string productIdentifier, string refundRequestStatus)
+        {
+            RefundRequestStatus status;
+            switch (refundRequestStatus?.ToUpperInvariant())
+            {
+                case "SUCCESS":
+                    status = RefundRequestStatus.Success;
+                    break;
+                case "USERCANCELLED":
+                case "USER_CANCELLED":
+                    status = RefundRequestStatus.UserCancelled;
+                    break;
+                default:
+                    status = RefundRequestStatus.Error;
+                    break;
+            }
+            
+            _storedCallbacks?.OnRefundRequestCompleted?.Invoke(
+                new RefundRequestCompletedEventArgs(productIdentifier, status)
+            );
+        }
+
+        private void OnManagementOptionSelected(string option, string url)
+        {
+            CustomerCenterManagementOption optionEnum;
+            switch (option?.ToLowerInvariant())
+            {
+                case "cancel":
+                    optionEnum = CustomerCenterManagementOption.Cancel;
+                    break;
+                case "custom_url":
+                    optionEnum = CustomerCenterManagementOption.CustomUrl;
+                    break;
+                case "missing_purchase":
+                    optionEnum = CustomerCenterManagementOption.MissingPurchase;
+                    break;
+                case "refund_request":
+                    optionEnum = CustomerCenterManagementOption.RefundRequest;
+                    break;
+                case "change_plans":
+                    optionEnum = CustomerCenterManagementOption.ChangePlans;
+                    break;
+                default:
+                    optionEnum = CustomerCenterManagementOption.Unknown;
+                    break;
+            }
+            
+            _storedCallbacks?.OnManagementOptionSelected?.Invoke(
+                new ManagementOptionSelectedEventArgs(optionEnum, url)
+            );
+        }
+
+        private void OnCustomActionSelected(string actionId, string purchaseIdentifier)
+        {
+            _storedCallbacks?.OnCustomActionSelected?.Invoke(
+                new CustomActionSelectedEventArgs(actionId, purchaseIdentifier)
+            );
+        }
+
+        private class CallbacksProxy : AndroidJavaProxy
+        {
+            private readonly AndroidCustomerCenterPresenter _owner;
+
+            public CallbacksProxy(AndroidCustomerCenterPresenter owner) : base("com.revenuecat.purchasesunity.ui.RevenueCatUI$CustomerCenterCallbacks")
```

Comment:
It's weird to have a public constructor and methods for a private class 😅. Should it be internal?

Reply:
the methods need to stay public so the JNI bridge can call them. The class can stay private though. I think that's the most encapsulated we can go